File: Java programming/problems on file handling/problem12.java
/*
12. Write a program which accept file name which contains information of student and isplay only names of students. #include<stdio.h>
	
	Author : Milind Nalawade
	
	Date : 20/08/2021
	
	Language : Java 14.0.2
	
*/
import java.lang.*;
import java.util.*;
import java.io.Serializable;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.io.ObjectInputStream;
import java.io.FileInputStream;
import java.lang.Exception;
class Info implements Serializable
{
	private String name;
	private int id;
	private int marks;

	public void Name(String n)
	{
		this.name = n;
	}
	public String getName()
	{
		return name;
	}

	public int getid() {
		return id;
	}

	public void setid(int ID) {
		this.id = ID;
	}

	public int getmarks() {
		return marks;
	}

	public void setmarks(int Marks) {
		this.marks = Marks;
	}

	@Override
	public String toString()
	{
		return " Name=" + this.name ;
	}

}



class Student
{
	static void insert(String filename,ArrayList<Info>ali)
	{
		try
		{
			FileOutputStream fos=new FileOutputStream(filename);
			if(fos==null)
			{
				System.out.print("\n Unable to open file \n");
			}
			else
			{
				System.out.print("\n File successfully opened\n");
			}
			ObjectOutputStream oos=new ObjectOutputStream(fos);
			oos.writeObject(ali);
			System.out.print("\nDone\n");
			
		}
		catch(IOException e)
		{
			e.printStackTrace();
		}
	}
	
	static void Display(String filename)
	{
		try
		{
			FileInputStream fis=new FileInputStream(filename);
			ObjectInputStream ois=new ObjectInputStream(fis);
			Info iobj=null;
			ArrayList<Info> ali=new ArrayList<>();
			ali=(ArrayList<Info>)ois.readObject();

			for(int i=0;i<ali.size();i++)
			{
				System.out.println(" "+ali.get(i).getName());
			}
		}
		catch (Exception e)
		{
			
		}		
	}
}

class problem12
{
	public static void main(String argv[])
	{
		Scanner sobj=new Scanner(System.in);
		System.out.print("\n Enter file name : ");
		String filename=sobj.nextLine();
		Student obj=new Student();
		System.out.print("\n Enter number of students : ");
		int size=sobj.nextInt();
		ArrayList<Info> ali=new ArrayList<>();
		for(int i=0;i<size;i++)
		{
			System.out.println();
			System.out.print(" Enter name of student : ");
			sobj.nextLine();
			String name=sobj.nextLine();
			System.out.println();
			System.out.print(" Enter id of student : ");
			int ID=sobj.nextInt();
			System.out.println();
			System.out.print(" Enter marks of student : ");
			int Marks=sobj.nextInt();
			Info iobj=new Info();
			iobj.Name(name);
			iobj.setid(ID);
			iobj.setmarks(Marks);
			ali.add(iobj);
		}
		obj.insert(filename,ali);
		System.out.print(" \n Names of the students are : \n\n");
		obj.Display(filename);
	}
}

File: Python Programming/problems on file handling/problem12.py
'''
12. Write a program which accept oinformation of students from user and write that information into the file. 
		
	   Author : Milind Nalawade

	   Date : 02/09/2021

	   Language : Python 3.9.0

'''
import os
import pickle
class StudentInfo:
	def __init__(self,Name,Id,Marks):
		self.name=Name
		self.id=Id
		self.marks=Marks

class Student:	
	def Insert(self,filename,stdobj):
		fd=open(filename,"ab")
		if fd==None:
			print("\n Unable to open the file \n")
		else:
			print("\n file opened successfully \n")
			pickle.dump(stdobj,fd);
		fd.close()
	
	def LoadData(self,filename):
		print()
		fd=open(filename,'rb')
		while fd:
			try :
				list=pickle.load(fd);
				print(list.name);
			except EOFError:
				break;
				
#=============
#Entry point function
#=============
def main():
	
	filename=input("\n Enter file name : ")
	size=int(input("\n Enter number of student : "))
	sobj=Student()
	
	for i in range(size):
		name=input("\n Enter student's name : ")
		id=int(input("\n Enter student's id : "))
		marks=int(input("\n Enter student's marks : "))
		SI=StudentInfo(name,id,marks)
		sobj.Insert(filename,SI)
	sobj.LoadData(filename)
	
#=====
#starter
#=====
if __name__=="__main__":
	main()